packages/api-client-core/src/exchanges/cacheExchange.ts

```diff
@@ -0,0 +1,237 @@
+/**
+ * This file was copied from: https://github.com/urql-graphql/urql/blob/25d114d25807f0676dbf453732602753279ba0db/packages/core/src/exchanges/cache.ts
+ * Any meaningful changes are documented with comments starting with "GADGET:"
+ */
```

Comment:
We can keep this comment, but should do one of two things to adhere to the MIT license: either copy https://github.com/urql-graphql/urql/blob/main/LICENSE into this directory (maybe we put it in a subdirectory so the license is properly scoped to just the files that matter) or we copy the LICENSE contents into the top of all files we copied.